### PR TITLE
reused the existing copy paste contoller and set a type flag

### PR DESCRIPTION
**Pull request by willdot**
PBI: 69704

Managed to reuse the existing copy paste controller here as I figured out a cleaner way to do it. 

If you're happy with the implementation, I'll take another look at reusing it for the sent on copy paste i did yesterday.

### CODE REVIEW

File: src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs [hunk 1]
```diff
@@ -40,10 +40,15 @@ public virtual async Task<ActionResult> Index(Guid returnId, Guid aatfId, Guid s
...
-                    SchemeId = schemeId,
-                    SchemeName = Task.Run(() => cache.FetchSchemePublicInfoBySchemeId(schemeId)).Result.Name,
-                    AatfName = Task.Run(() => cache.FetchAatfData(@return.ReturnOperatorData.OrganisationId, aatfId)).Result.Name
+                    AatfName = Task.Run(() => cache.FetchAatfData(@return.ReturnOperatorData.OrganisationId, aatfId)).Result.Name,
+                    Type = obligatedType
                 };
+
+                if (obligatedType == ObligatedType.Recieved)
+                {
+                    viewModel.SchemeId = schemeId;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs]: Add unit tests to test these properties for when ObligatedType.Received

File: src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs [hunk 2]
```diff
@@ -64,6 +69,11 @@ public virtual async Task<ActionResult> Index(ObligatedValuesCopyPasteViewModel
                 }
             }
 
+            if (viewModel.Type == ObligatedType.Reused)
+            {
+                return await Task.Run<ActionResult>(() => AatfRedirect.ObligatedReused(viewModel.ReturnId, viewModel.AatfId));
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs]: Add unit tests to test for this redirect when reused

File: src/EA.Weee.Core/AatfReturn/ObligatedType.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace EA.Weee.Core.AatfReturn
+{
+    public enum ObligatedType
+    {
+        Recieved,
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Core/AatfReturn/ObligatedType.cs]: Typo :) Received

File: src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs (hunk 1 above)
Comment (willdot) [reply src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs]: The existing test tested with the scheme data, so I added a unit test to test for when its reused and made sure the scheme properties on the VM were not set correctly.

File: src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs (hunk 2 above)
Comment (willdot) [reply src/EA.Weee.Web/Areas/AatfReturn/Controllers/ObligatedValuesCopyPasteController.cs]: Added tests for both.

File: src/EA.Weee.Core/AatfReturn/ObligatedType.cs (hunk 3 above)
Comment (willdot) [reply src/EA.Weee.Core/AatfReturn/ObligatedType.cs]: I hate I before E.....

**Pull request merged**